### PR TITLE
SHM description

### PR DESCRIPTION
**Pull request by MVolobueva**
add SHM

### CODE REVIEW

**Review by Alexander230 — APPROVED** [review]

File: vignettes/web_only/BCRpipeline.Rmd [hunk 1]
```diff
@@ -44,9 +44,15 @@ The pipeline involves five steps:
...
+
 5. **Phylogenetic analysis.**
 
-    This final step provides phylogeny reconstruction and trunk length calculation (by running the PHYLIP package).
+    This step provides phylogeny reconstruction and trunk length calculation (by running the PHYLIP package).
+    
+
+6. **Somatic hypermutation analysis.**
+
+    At this step, clonotype and germline sequences are compared to count the number of mutations.
```
Comment (zozozoe) [inline vignettes/web_only/BCRpipeline.Rmd:55]: At this step, we compare clonotype and germline sequences to detect and count the number of mutations in smth.

File: vignettes/web_only/BCRpipeline.Rmd [hunk 2]
```diff
@@ -233,14 +239,81 @@ bcr$full_clones$Cluster %>% unique()
...
 We have calculated a trunk length for each cluster:
 
 ```{r example 13}
-bcr$full_clones$Trunk.Length
+bcr$full_clones[ , c('Cluster', 'Trunk.Length') ]
 ```
+
+# Somatic hypermutation analysis
+
+Somatic hypermutation analysis is extremely useful for BCRs. The somatic hypermutation rate allows us to estimate the maturation of repertoire and figure out what kind of mutation contributes to the emergence of high-affinity antibodies.
```
Comment (zozozoe) [inline vignettes/web_only/BCRpipeline.Rmd:257]: 1. omit "extremely" or clarify
2. The rate of somatic hypermutation allows us to estimate repertoire maturation and detect the type of mutation contributing to the emergence on high affinity antibodies.

File: vignettes/web_only/BCRpipeline.Rmd [hunk 3]
```diff
@@ -233,14 +239,81 @@ bcr$full_clones$Cluster %>% unique()
...
+bcr_data <- bcrdata$data
+
+shm_data <- bcr_data %>%
+  seqCluster(seqDist(bcr_data), .fixed_threshold = 3) %>%
+  repGermline(.threads = 2) %>%
+  repAlignLineage(.min_lineage_sequences = 2, .align_threads = 2, .nofail = TRUE) %>%
+  repClonalFamily(.threads = 2, .nofail = TRUE) %>%
+  repSomaticHypermutation(.threads = 2, .nofail = TRUE)
+```
+The function `repSomaticHypermutation()` takes germline V,  germline J sequences and clonotype V and clonotypes J sequences and alignes germline and clonotype sequences to calculate mutations.
```
Comment (zozozoe) [inline vignettes/web_only/BCRpipeline.Rmd:271]: The function `repSomaticHypermutation()` takes V and J germline sequences and V and J clonotype sequences as an input. Then the function aligns germline and clonotype sequences to detect and calculate occurring mutations.

File: vignettes/web_only/BCRpipeline.Rmd [hunk 4]
```diff
@@ -233,14 +239,81 @@ bcr$full_clones$Cluster %>% unique()
...
+shm_data$full_clones$J.germline.nt[1]
+
+# the example of clonotype V sequence
+cols <- c('FR1.nt', 'CDR1.nt', 'FR2.nt', 'CDR2.nt', 'FR3.nt')
+apply(shm_data$full_clones[1,][ , cols ] , 1 , paste , collapse = "" )[[1]]
+
+# the example of clonotype J sequence
+shm_data$full_clones$FR4.nt[1]
+```
+The example of alignment between germline and clonotype V sequences:`
```
Comment (zozozoe) [inline vignettes/web_only/BCRpipeline.Rmd:289]: Example: aligning germline and clonotype V sequences:

File: vignettes/web_only/BCRpipeline.Rmd [hunk 5]
```diff
@@ -233,14 +239,81 @@ bcr$full_clones$Cluster %>% unique()
...
+
+```{r example 16}
+image(shm_data$full_clones$Germline.Alignment.V[[1]], grid = TRUE)
+```
+
+The example of alignment between germline and clonotype J sequences:`
```
Comment (zozozoe) [inline vignettes/web_only/BCRpipeline.Rmd:295]: Example: aligning germline and clonotype V sequences:

File: vignettes/web_only/BCRpipeline.Rmd [hunk 6]
```diff
@@ -233,14 +239,81 @@ bcr$full_clones$Cluster %>% unique()
...
+```
+
+The number of mutations for each clonotype sequence:
+
+```{r example 18}
+cols <- c('Clone.ID', 'Substitutions', 'Insertions', 'Deletions', 'Mutations')
+shm_data$full_clones[ , cols ]
+```
+
+Then you could easily estimate mutation rate:
```
Comment (zozozoe) [inline vignettes/web_only/BCRpipeline.Rmd:308]: Then you could easily estimate the mutation rate:

**Review by zozozoe — COMMENTED** [review]
Minor grammar edits. Need clarification for the SHM general descriptions

File: vignettes/web_only/BCRpipeline.Rmd (hunk 2 above)
Comment (zozozoe) [reply vignettes/web_only/BCRpipeline.Rmd:257]: The rate of somatic hypermutation allows us to estimate repertoire maturation and detect the type of mutation contributing to the emergence on high affinity antibodies. This makes somatic hypermutation analysis a valuable asset for B-cell repertoire analysis.

Comment (zozozoe) [reply vignettes/web_only/BCRpipeline.Rmd:257]: The rate of somatic hypermutation allows us to estimate repertoire maturation and detect the type of mutation contributing to the emergence of high affinity antibodies.

File: vignettes/web_only/BCRpipeline.Rmd (hunk 5 above)
Comment (zozozoe) [reply vignettes/web_only/BCRpipeline.Rmd:295]: Example: aligning germline and clonotype J sequences:

**Pull request merged**